### DISCUSSION TITLE
How to use the efficient pretrained model in a closed system with no internet access

### DISCUSSION THREAD

**Discussion by JunMa11** [Project-MONAI/MONAI#3374] [Q&A] [answered]
Dear MONAI Team,

Thanks for the awesome work.

I want to use the pre-trained efficient model on a closed server without internet access.
I have manually downloaded and uploaded the [pretrained models](https://github.com/lukemelas/EfficientNet-PyTorch/blob/7e8b0d312162f335785fb5dcfa1df29a75a1783a/efficientnet_pytorch/utils.py?_pjax=%23js-repo-pjax-container%2C%20div%5Bitemtype%3D%22http%3A%2F%2Fschema.org%2FSoftwareSourceCode%22%5D%20main%2C%20%5Bdata-pjax-container%5D#L556) to the server.

My current code is 
```
from monai.networks.nets import EfficientNetBN
model = EfficientNetBN("efficientnet-b0", pretrained=Train, spatial_dims=2, in_channels=3, num_classes=num_class).to(device)

outputs = model(inputs)
```

How should I modify the code to set the path to the pre-trained models and load them?

Kindest regards,
Jun

**Comment by Nic-Ma** [comment]
Hi @JunMa11 ,

Thanks for your feedback.
I feel this is an interesting topic, @yiheng-wang-nv could you please help think about it?
Maybe you can improve the `pretrained` logic your previously added to allow local path weights?

Thanks in advance.

**Comment by wyli** [comment, marked as answer]
the default dir is from `torch.hub`, you can put the downloaded .pth file in the default path https://github.com/pytorch/pytorch/blob/e1c449ff342ee00a4b25e1651bd08a41b7e9fda4/torch/hub.py#L568:
```
>>> import torch
>>> import os
>>> os.path.join(torch.hub.get_dir(), "checkpoints")
```

**Reply by Nic-Ma** [reply]
Good idea!

**Reply by JunMa11** [reply]
Thank you so much.
